packages/starlight/utils/path.ts

```diff
@@ -50,3 +50,14 @@ export function ensureHtmlExtension(path: string) {
 	}
 	return ensureLeadingSlash(path);
 }
+
+export function splitQueryAndFragment(url: string) {
+	const [urlWithQuery, hash] = url.split('#');
+	const [urlBase, queryString] = (urlWithQuery || '').split('?');
+
+	return {
+		base: urlBase || '',
+		query: queryString ? `?${queryString}` : '',
+		fragment: hash ? `#${hash}` : '',
+	};
```

Comment:
We could potentially lean on native URL parsing here rather than rolling our own. What do you think? In this example, I’d suggest returning the properties exactly as they are named in the web standard API: `pathname` rather than `base`, `search` rather than `query` (`query` in the `URL` object excludes the `?`), and `hash` rather than `fragment`:

```suggestion
        const { pathname, search, hash } = new URL(url, 'https://example.com');
	return { pathname, search, hash };
```

If the `example.com` looks a bit funny here, the way this works is that it resolves the first argument relative to the second, so for relative arguments (e.g. `/foo`) we need a base to resolve against otherwise it can’t construct a full URL. We’re only using the origin at all, so `example.com` is ignored in output here.

Reply:
@delucis - Thanks for all the input!  Regarding above, the reason I didn't use `URL` for the parsing is due to #2647 as the favicon can be absolute or relative.  I considered detecting if it is absolute or internal and then using `URL` but that seemed rather fragile in terms of approach.  Additionally, to detect absolute vs relative, my options were to add a dependency to do the detection or roll my own and didn't feel that either was ideal.  Since we really need to only grab `base` regardless of whether its absolute or relative, I landed on the approach I took.  Happy to adjust, just need to make sure that we handle absolute and relative in this case - let me know your thoughts.

Reply:
I think that’s OK because the code doesn’t use the data this function returns for anything other than checking the extension.

Reply:
Sorry, just wanted to clarify what you mean by "OK" - do you mean leave as I have it or change to detect absolute vs relative URL and then use URL accordingly?

Reply:
- Use `URL` directly
- No need to detect absolute vs relative because we’re not using this function to get the final URL at all, only to extract the extension.

Reply:
TIL -  `URL` constructor will ignore `base` when `url` is absolute so `new URL('https://starlight.build.dev/favicon.svg', 'https://www.example.com') will succeed.

I had assumed (I should have actually tested it ☹️ ) that it will fail, hence all comments above, sorry!

I'll adjust to use URL and update PR, thanks!

Reply:
Eliminated `splitQueryAndFragment` util function as it's no longer necessary given use of `URL` constructor.